SU-HSDO/suhumsci
# STN-590:  Email and URL refinement with word break

# READY FOR REVIEW

## Summary
Currently, if an email or URL is longer than ~18 characters it will flow beyond the container of a group block on a node detail page or when raised cards is used. This was also present if the email was not a link and was just listed as plain text.

This PR alleviates those issues for those areas, allowing the url or the email to word break on that line and stay within the container of the block or card.

Example in a group block: https://buddhiststudies.stanford.edu/people/egil-gil-fronsdal
Example with raised cards (see responsive): https://feminist-prod.stanford.edu/what-we-do/contact-us

JIRA Card:[ STN-590](https://sparkbox.atlassian.net/browse/STN-590)

## Need Review By (Date)
11/5/20

## Urgency
medium

## Steps to Test
1. Run `npm run test`
2. To test and validate this fix for these areas you may have to add some content to different locations and test them.
3. To test the group block on a people page you can go to http://sparkbox-sandbox.suhumsci.loc/people/ran-abramitzky
- Update the email address for this page to something extremely long and you should see it wrap rather than run outside of the container. Example of what it should do below.
![image](https://user-images.githubusercontent.com/9355698/98154940-854f4780-1ea3-11eb-8952-eda242c61f0e.png)
4. To test the raised cards, go to http://sparkbox-sandbox.suhumsci.loc/qa/qa-raised-postcards
- You will want to scroll down to the vertical cards and add long url's, long emails (link and plain text) and verify that the content of those cards stays within their description blocks.  This was addressed to the card description properties themselves including for vertical linked cards which also could have the same problem if a long email was added or a long more link name is added for the traditional theme more links on the cards. Example of what it should do below.
![image](https://user-images.githubusercontent.com/9355698/98155331-10304200-1ea4-11eb-9a92-7922ef78c007.png)
5. To test Event detail pages, it was noted that in the columned block at the bottom of the event detail pages long emails were running into the contact category to the right of the email.  This fix should force the email to wrap within it's given category area. Go to http://sparkbox-sandbox.suhumsci.loc/events/economic-risk-within-household-and-voting-radical-right
- You will want to update the email to be a longer email for testing purposes and it should look like this when it wraps.
![image](https://user-images.githubusercontent.com/9355698/98155966-1e329280-1ea5-11eb-8787-94dbb9c758c6.png)
6. All of these fixes should work the same for both Traditional and Colorful themes. The biggest testing difference is the more links for cards on the Traditional theme, those have text added and should also wrap if a very long word is added without natural word breaks for wrapping.


## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
